dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -130,14 +141,22 @@ public MetadataResource(SessionFactory sessionFactory, DockstoreWebserviceConfig
     @Operation(summary = "List all available workflow, tool, organization, and collection paths.", description = "List all available workflow, tool, organization, and collection paths. Available means published for tools/workflows, and approved for organizations and their respective collections. NO authentication")
     @ApiOperation(value = "List all available workflow, tool, organization, and collection paths.", notes = "List all available workflow, tool, organization, and collection paths. Available means published for tools/workflows, and approved for organizations and their respective collections.")
     public String sitemap() {
-        List<String> urls = new ArrayList<>();
+        try {
+            SortedSet<String> sitemap = sitemapListener.getCache().get("sitemap", this::getSitemap);
```

Comment:
See other comment, but if sitemapListener is null this will fail. Which is why I think you should throw an exception if you can't find it in the constructor.